code/test/general.js

```diff
@@ -0,0 +1,2306 @@
+    const { testerc20 } = await TestUtil.makeTokens(accounts[0])
+    let aliceBalanceERC20
+    let aliceWalletBalanceERC20
+    // transfer ERC20 tokens from accounts[0] (which owns the tokens) to alices wallet
+    await testerc20.transfer(alice.wallet.address, 1000, { from: accounts[0] })
+    aliceCurrentState = await TestUtil.getONEWalletState(alice.wallet)
+    aliceBalanceERC20 = await testerc20.balanceOf(alice.wallet.address)
+    assert.equal(1000, aliceBalanceERC20, 'Transfer of 1000 ERC20 tokens to alice.wallet succesful')
+    aliceWalletBalanceERC20 = await alice.wallet.getBalance(ONEConstants.TokenType.ERC20, testerc20.address, 0)
+    assert.equal(1000, aliceWalletBalanceERC20, 'Transfer of 1000 ERC20 tokens to alice.wallet checked via wallet succesful')
+    console.log('Funded ERC20')
+
+    // Begin Tests
+    let testTime = Date.now()
+
+    testTime = await TestUtil.bumpTestTime(testTime, 60)
+    // create randomSeed
+    let randomSeed = new Uint8Array(new BigUint64Array([0n, BigInt(testTime)]).buffer)
+    // recover Alices wallet
+    //
+    await TestUtil.transactionExecute(
+      {
+        wallet: alice,
+        operationType: ONEConstants.OperationType.RECOVER,
+        randomSeed,
+        testTime
+      }
+    )
+    // Update alice and bob's current State
+    aliceCurrentState = await TestUtil.getONEWalletState(alice.wallet)
+    // Alice Items that have changed - nonce, lastOperationTime, commits, trackedTokens
+    // nonce
+    let nonce = await alice.wallet.getNonce()
+    assert.notEqual(nonce, aliceOldState.nonce, 'alice wallet.nonce should have been changed')
+    assert.equal(nonce.toNumber(), aliceOldState.nonce + 1, 'alice wallet.nonce should have been changed')
+    aliceOldState.nonce = nonce.toNumber()
+    // lastOperationTime
+    let lastOperationTime = await alice.wallet.lastOperationTime()
+    assert.notStrictEqual(lastOperationTime, aliceOldState.lastOperationTime, 'alice wallet.lastOperationTime should have been updated')
+    aliceOldState.lastOperationTime = lastOperationTime.toNumber()
+    // spendingState
+    let spendingState = await alice.wallet.getSpendingState()
+    assert.equal(spendingState.spentAmount, (ONE_CENT / 2).toString(), 'alice wallet.spentAmount should have been changed')
+    aliceOldState.spendingState.spentAmount = spendingState.spentAmount
+    assert.notEqual(spendingState.lastSpendingInterval, '0', 'alice wallet.spentAmount should have been changed')
+    aliceOldState.spendingState.lastSpendingInterval = spendingState.lastSpendingInterval
+    // commits
+    let allCommits = await alice.wallet.getAllCommits()
+    assert.notDeepEqual(allCommits, aliceOldState.allCommits, 'alice wallet.allCommits should have been updated')
+    aliceOldState.allCommits = allCommits
+    // check alice
+    await TestUtil.checkONEWalletStateChange(aliceOldState, aliceCurrentState)
+  })
+
+  // ==== DISPLACE =====
```

Comment:
remove tests below that are not implemented / not ready for production or auditing / are boilerplates? Add them later in separate PR when they are production ready